AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/Statement+Model.swift

```diff
@@ -88,9 +89,39 @@ extension Statement: StatementModelConvertible {
                 from: value,
                 fieldType: field.type
             )
-            modelDictionary.updateValue(modelValue, forKeyPath: column)
+
+            // Check if the value for the primary key is `nil`. This is when an associated model does not exist.
+            // To create a decodable `modelDictionary` that can be decoded to the Model types, the entire
+            // object at this particular key should be set to `nil`. The following code does that by dropping the last
+            // path from the column, for example given "blog.id" `column` has a nil `modelValue`, then store the
+            // keypath in `skipColumns`, which will be used to set modelDictionary["blog"] to nil later on.
+            //
+            // `skipColumns` keeps track of these scenarios. At this point in the code we cannot make an assumption
+            // about the ordering of the columns, it may handle "blog.description", then "blog.id", then "blog.title".
+            // The code will perform the following:
+            //  1. set ["blog"]["description"] = `nil`, because it has not encountered `nil` primary key
+            //  2. store skipColumn["blog"], because the value is the primary key and is nil
+            //  3. skip setting modelDictionary["blog"]["title"], because `skipColumn` has been set for "blog".
+            if field.isPrimaryKey && modelValue == nil {
```

Comment:
i am wondering how this will work in case of a composite primary key where there would be an array of fields for primary keys and a custom primary key ""key1"#"key2".." with a column @ @ primarykey

Reply:
I compared the two PRs, this one and the CPK one, in this PR
```
public var isPrimaryKey: Bool {
        return name == "id"
    }
```

In CPK PR
```swift
@available(*, deprecated, message: "Use the primaryKey member of the schema")
    public var isPrimaryKey: Bool {
        return attributes.contains { $0 == .primaryKey }
    }
```

It appears in the CPK PR, it is still checking the deprecated isPrimaryKey from the field for creating lazy lists on the associations (Under the comment `// create lazy list for "many" relationships`, if let id = modelValue as? String, field.isPrimaryKey`). if there is one primaryKey on the 1 model then it will create a lazy list for each association. If there is a composite key, it will create a lazy list for each association for each time it finds a key that is part of the composite key. the lazy list would be overwritten by the latest lazy list on each component of the composite key, thus the associated id stored in the composite key is only part of the id that can be used for lazy laoding. I've opened a comment over in the CPK PR to fix this: https://github.com/aws-amplify/amplify-ios/pull/1752/files#r900313614

Now back to this check that we're introducing here, if any of the fields is a primary key and the value is `nil`, we will mark the entire model as should be skipped / set to `nil`. This will still work in this case of a composite key since if any of the components of the composite key has a `nil` value, the code will evaluate to true, and run the code at lines 106 and 107 twice. It still works as we expect it to. Until we fully deprecate the field, we can still continue to use it, however should migrate to the CPK's latest way to get the primary key from the schema. I can't exactly do that in this PR since that would mean we would have to ship the CPK PR first.